crates/primitives/src/batch.rs

```diff
@@ -210,6 +219,29 @@ impl Checkpoint {
 
         hash::raw(&buf)
     }
+
+    pub fn sidecar(&self) -> &CheckpointSidecar {
+        &self.sidecar
+    }
+}
+
+#[derive(
+    Clone, Debug, PartialEq, Eq, Arbitrary, BorshDeserialize, BorshSerialize, Deserialize, Serialize,
+)]
+pub struct CheckpointSidecar {
+    /// Chainstate at the end of this checkpoint's epoch.
+    /// Note: using Vec<u8> instead of Chainstate to avoid circular dependency with strata_state
+    chainstate: Vec<u8>,
+}
+
+impl CheckpointSidecar {
+    pub fn new(chainstate: Vec<u8>) -> Self {
+        Self { chainstate }
+    }
+
+    pub fn chainstate(&self) -> &[u8] {
+        &self.chainstate
+    }
 }
```

Comment:
I suggested the name `LazyChainstateBlob` in [this ticket](https://alpenlabs.atlassian.net/browse/STR-1055) that I shared earlier.

Reply:
Why `Lazy` though? Also, something like `CheckpointSidecar` or `CheckpointExtra` indicates that we are not restricted to posting chainstate only. This seems descriptive enough to me.

Reply:
To emphasize that this is *not* how we intend to do it long term.  It'll look a lot different.

Reply:
To be honest, naming is hard, and very very subjective. For me `Lazy` by _no means_ implies that "we don't intend to do this long term". For me `Lazy` means something that is deferred until it is actually needed.

Let's not please debate over petty names.